src/System.Net.Security/src/System/Net/StreamFramer.cs

```diff
@@ -175,13 +175,8 @@ private void ReadFrameCallback(IAsyncResult transportResult)
             {
                 ReadFrameComplete(transportResult);
             }
-            catch (Exception e)
+            catch (Exception e) when (!(e is OutOfMemoryException))
             {
-                if (e is OutOfMemoryException)
-                {
-                    throw;
-                }
-
```

Comment:
Why not:
```C#
catch (OutOfMemoryException)
{
    throw;
}
catch(Exception e)
{
 …
```
It seems more natural to use the exception-type-matching mechanism than to have a filter that only does exception-type-matching.